xcdat/regridder/regrid2.py

```diff
@@ -0,0 +1,547 @@
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+import xarray as xr
+
+from xcdat.regridder.base import BaseRegridder
+
+
+def extract_bounds(bounds: xr.DataArray) -> Tuple[xr.DataArray, xr.DataArray]:
+    """
+     Extract lower and upper bounds from an axis.
+
+     Parameters
+     ----------
+     bounds : xr.DataArray
+         Dataset containing axis with bounds.
+
+     Returns
+     -------
+    Tuple[xr.DataArray, xr.DataArray]
+         A tuple containing the lower and upper bounds for the axis.
+    """
+    if bounds[0, 0] < bounds[0, 1]:
+        lower = bounds[:, 0]
+        upper = bounds[:, 1]
+    else:
+        lower = bounds[:, 1]
+        upper = bounds[:, 0]
+
+    return lower, upper
+
+
+def map_latitude(src: xr.DataArray, dst: xr.DataArray) -> Tuple[List, List]:
+    """
+    Map source to destination latitude.
+
+    Parameters
+    ----------
+    src : xr.DataArray
+        DataArray containing the source latitude bounds.
+    dst : xr.DataArray
+        DataArray containing the destination latitude bounds.
+
+    Returns
+    -------
+    Tuple[List, List]
+        A tuple of cell mappings and cell weights.
+    """
+    src_south, src_north = extract_bounds(src)
+    dst_south, dst_north = extract_bounds(dst)
+
+    mapping = []
+    weights = []
+
+    for i in range(dst.shape[0]):
+        contrib = np.where(
+            np.logical_and(src_south < dst_north[i], src_north > dst_south[i])
+        )[0]
+
+        mapping.append(contrib)
+
+        north_bounds = np.minimum(dst_north[i], src_north[contrib])
+        south_bounds = np.maximum(dst_south[i], src_south[contrib])
+
+        weight = np.sin(np.deg2rad(north_bounds)) - np.sin(np.deg2rad(south_bounds))
+
+        weights.append(weight.values.reshape(contrib.shape[0], 1))
+
+    return mapping, weights
+
+
+def pertub(value):
```

Comment:
Missing type annotation for `value` and return type

Reply:
This is resolved in the latest.